### PR TITLE
Added username choice

### PR DESCRIPTION
**Pull request by giommok**
Implemented the **username choice**

Features:
- On the first sign-in the user is asked to select his username.
- If on the first sign-in the user doesn't insert a username, he will be asked to after each login, as the user will be registered only after he selects a correct username.
- A logged and registered user will be able to change his username.
- A user won't be able to choose a used username or the username he already has (if he has one).

Added tests:
- Correct display if no user is signed.
- Correct display of EditText hint text.
- Correct behaviour of synchronizeUsername() method.

### CODE REVIEW

File: app/src/androidTest/java/com/github/giommok/softwaredevproject/AccountTest.java [hunk 1]
```diff
@@ -27,6 +24,20 @@ public void noAccountTest() {
         assertEquals(account.getId(), "null");
         assertEquals(account.getProviderId(), "null");
         assertEquals(account.getPhotoUrl(), Uri.EMPTY);
+    }
 
+    @Test
+    public void synchronizeUsernameTest() throws InterruptedException {
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/giommok/softwaredevproject/AccountTest.java:30]: Add small comment containing test description

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java [hunk 2]
```diff
@@ -58,14 +60,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            Database.isPresent("users", "email", account.getEmail(), this::setUI, this::setUsernameChoiceUI);
+        }
     }
 
     public void signInButton(View view) {
         Intent signInIntent = mGoogleSignInClient.getSignInIntent();
         startActivityForResult(signInIntent, RC_SIGN_IN);
     }
 
+    public void changeUsernameButton(View view) {
```
Comment (bgabriel998) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java:75]: Javadoc comment missing

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java [hunk 3]
```diff
@@ -58,14 +60,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        setUsernameChoiceUI();
+    }
+
+    public void submitUsernameButton(View view) {
```
Comment (bgabriel998) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java:79]: Javadoc comment missing

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java [hunk 4]
```diff
@@ -58,14 +60,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        Log.d("CURRENT_USERNAME", "onSubmit: " + currentUsername);
+        if(username.equals(currentUsername)) {
+            Log.d("CURRENT_USERNAME", "onSubmit: EQUAL_CASE");
+            Toast toast = Toast.makeText(getApplicationContext(), "You can't choose the username you already have!", Toast.LENGTH_SHORT);
+            toast.show();
+        }
+        else Database.isPresent("users", "username", username, () -> usernameAlreadyPresent(username) , () -> registerUser(username));
+    }
+
     public void signOutButton(View view) {
```
Comment (bgabriel998) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java:91]: Javadoc comment missing

**Review by bgabriel998 — APPROVED** [review]
Few comments are missing. Otherwise everything is fine

**Comment by giommok** [comment]
> Few comments are missing. Otherwise everything is fine

Added comments

**Review by magangloff — APPROVED** [review]
Looks all good.

**Pull request merged**